### DISCUSSION TITLE
Error while inserting new row [Jun 10]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2418] [Discord Threads]
> 🙋‍♂️ **vibrantium**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### can you help us with the error message?  
#  
> 🙋‍♂️ **vibrantium**  
#### Sure! It's super long though :( but first of all is the "await axios.request(create)" a correct sentence? I've seen people use "await axios.post"  
#  
> 👨‍🦲 **dstala**  
#### axios.request(create) is fine. you will have to replace fields with data: {}  
#  
> 👨‍🦲 **dstala**  
#### ```const options2 = {   
####   method: 'POST',   
####   url: 'http://localhost:8080/api/v1/db/data/noco/p_wiukyreqe052rf/Producer/',   
####   data: { Name: "x1", Notes: "x1" },   
####   headers: {   
####     'xc-auth': authToken   
####   }   
#### };   
####    
#### axios.request(options2).then(function (response) {   
####   console.log(response.data);   
#### }).catch(function (error) {   
####   console.error(error);   
#### });```  
#  
> 👨‍🦲 **dstala**  
#### this worked for me.  
#  
> 👨‍🦲 **dstala**  
#### you can find ready sample code (mapped for read api) on clicking get api snippet on your right bottom of grid view window  
#  
> 👨‍🦲 **dstala**  
#### here  
![image0](https://cdn.discordapp.com/attachments/983951411802959912/983966127363346483/Screenshot_2022-06-08_at_10.59.37_AM.png)  
#  
> 🙋‍♂️ **vibrantium**  
#### Okay! I tried replacing "fields" to "data" instead & while the error message looks a lot shorter now, here's what it says:   
####    
#### ```TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received an instance of Object```   
####    
#### I did use a variable for the url, it works for when I'm querying/getting a row 🤔  
#  
> 👨‍🦲 **dstala**  
#### try dumping url content to confirm if all good  
#  
> 🙋‍♂️ **vibrantium**  
#### okay just did! pasted the url instead of using the variable where i stored it. still gave me the same TypeError message  
#  
> 👨‍🦲 **dstala**  
#### `${sheet_Users}` : can you dump this? to confirm if its object or a string  
#  
> 🙋‍♂️ **vibrantium**  
#### Wanted to clarify: you "dump" a variable by doing `console.log(variable)` right?  
#  
> 👨‍🦲 **dstala**  
#### yes  
#  
> 🙋‍♂️ **vibrantium**  
#### ah okay good let me share what I got  
#  
> 🙋‍♂️ **vibrantium**  
#### So this is the console log line:   
#### ```js   
#### console.log(`This is the line right before NocoDB inputting, sheet_Users looks like: ${sheet_Users}`)```   
####    
#### and this is how it looks like in my log:   
#### ```This is the line right before NocoDB inputting, sheet_Users looks like: http://heyheka.herokuapp.com/api/v1/db/data/noco/p_s828dsukkjkkq4/Heka - User Data/views/Heka - User Data/```  
#  
> 👨‍🦲 **dstala**  
#### there are spaces in name. try url: `${encodeURI(sheet_Users)}`,  
#  
> 🙋‍♂️ **vibrantium**  
#### still gave me the same eror  
#  
> 🙋‍♂️ **vibrantium**  
#### Also I just noticed that one of my fields is a single select field? Do I have to input my data differently or can I just write down my selection name?  
#  
> 👨‍🦲 **dstala**  
#### single select, option in quotes. multi select, comma separated string under quotes (example: "option-1, option-2")  
#  
> 👨‍🦲 **dstala**  
#### dump encoded URL value once. to confirm if its fine.  
#  
> 🙋‍♂️ **vibrantium**  
#### okay, I think I solved the problem! It lies in a completely different line of the coding, which I didn't expect. Thanks for helping me out, still!  
#  
> 🙋‍♂️ **vibrantium**  
#### Okay, tiny question that I'm going to ask here since it's such a small question: is rowId the one you get when you query/"get" a row & within the list, there's a key-pair value that goes like: `Id: number`?  
#  
> 👨‍🦲 **dstala**  
#### yes. you can also check it in UI by enabling 'Show System Fields"  
![image0](https://cdn.discordapp.com/attachments/983951411802959912/984048304444829746/Screenshot_2022-06-08_at_4.26.19_PM.png)  
#  
> 🙋‍♂️ **vibrantium**  
#### Cool! Thanks for answering. Also can we reorder field columns? 👀  
#  
> 👨‍🦲 **dstala**  
#### yeah! you can. use dotted matrix in the picture shown above to drag drop  
#  
> 👨‍🦲 **dstala**  
#### in similar manner, you can re-order tables & views as well  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/983951411802959912)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
